### PR TITLE
Add Linux ASAN/lint CI via GitHub Actions

### PR DESCRIPTION
**Pull request by jamessan**
This generalizes much of the CI code (mainly in naming) and adds a new GitHub actions workflow for Linux builds.  Currently only the ASAN and lint jobs from Travis have been implemented, but that should be a good starting point.

### CODE REVIEW

**Comment by jamessan** [comment]
Doesn't look like this will run in the neovim repo until it gets merged, but you can see the [results](https://github.com/jamessan/neovim/actions/runs/362115310) from my repo.

File: CMakeLists.txt [hunk 1]
```diff
@@ -322,10 +322,10 @@ if(HAS_DIAG_COLOR_FLAG)
   endif()
 endif()
 
-option(TRAVIS_CI_BUILD "Travis/sourcehut CI, extra flags will be set" OFF)
```
Comment (jamessan) [inline CMakeLists.txt:325]: The sr.ht builds are currently defining `-DTRAVIS_CI_BUILD` and I think we would need @justinmk to change that.

Comment (jamessan) [reply CMakeLists.txt:325]: Never mind, just found where those are defined.

File: ci/install.sh [hunk 2]
```diff
@@ -19,7 +19,7 @@ echo "Install neovim module for Python 2."
 CC=cc python2 -m pip -q install --user --upgrade pynvim
 
 echo "Install neovim RubyGem."
-gem install --no-document --pre neovim
+gem install --no-document --user-install --pre neovim
```
Comment (janlazo) [inline ci/install.sh:22]: Ruby tests are skipped. Either update `PATH`, run sudo, or setup rvm,rbenv,chruby.

```
/usr/local/lib/site_ruby/2.5.0/rubygems/installer.rb:715: warning: Insecure world writable dir /home/runner in PATH, mode 040777
WARNING:  You don't have /home/runner/.gem/ruby/2.5.0/bin in your PATH,
	  gem executables will not run.
```

Comment (jamessan) [reply ci/install.sh:22]: > Ruby tests are skipped. Either update `PATH`, run sudo, or setup rvm,rbenv,chruby.

[Fixed](https://github.com/neovim/neovim/pull/13297).  Thanks for noticing.

**Pull request merged**